js/graviex.js

```diff
@@ -0,0 +1,991 @@
+        //   "created_at": "2022-02-07T10:09:23+03:00",
+        //   "volume": "0.141",
+        //   "remaining_volume": "0.141",
+        //   "executed_volume": "0.0",
+        //   "trades_count": 0,
+        //   "trades": [],
+        //   "strategy": null
+        // }
+        const order = this.parseOrder (response);
+        const status = order['status'];
+        if (status === 'closed' || status === 'canceled') {
+            throw new OrderNotFound (this.id + ' ' + this.json (order));
+        }
+        return order;
+    }
+
+    async cancellAllOrders (side = undefined, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {
+        };
+        if (side !== undefined) {
+            request['side'] = side;
+        }
+        const response = await this.privatePostOrdersClear (this.extend (request, params));
+        const order = this.parseOrder (response);
+        const status = order['status'];
+        if (status === 'closed' || status === 'canceled') {
+            throw new OrderNotFound (this.id + ' ' + this.json (order));
+        }
+        return order;
+    }
+
+    async fetchClosedOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {
+            'order_by': 'desc',
+        };
+        let market = undefined;
+        if (symbol !== undefined) {
+            request['market'] = this.marketId (symbol);
+            market = this.market (symbol);
+        }
+        if (limit !== undefined) {
+            request['limit'] = limit; // default: 500
+        }
+        const response = await this.privateGetOrdersHistory (this.extend (request, params));
+        // [
+        //   {
+        //     "id": 287792646,
+        //     "at": 1643956089,
+        //     "side": "sell",
+        //     "ord_type": "limit",
+        //     "price": "0.141",
+        //     "avg_price": "0.0",
+        //     "state": "cancel",
+        //     "market": "ltcbtc",
+        //     "created_at": "2022-02-04T09:28:09+03:00",
+        //     "volume": "0.141",
+        //     "remaining_volume": "0.141",
+        //     "executed_volume": "0.0",
+        //     "trades_count": 0,
+        //     "strategy": null
+        //   }
+        // ]
+        return this.parseOrders (response, market, since, limit);
+    }
+
+    async createDepositAddress (code, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        const request = {
+            'currency': currency['id'].toLowerCase (),
+        };
+        const response = await this.privateGetGenDepositAddress (this.extend (request, params));
+        // "\"request_accepted\""
+        const parsedResponse = JSON.parse (response);
+        return this.parseDepositAddress (parsedResponse, currency);
+    }
+
+    nonce () {
+        return this.milliseconds ();
+    }
+
+    encodeParams (params) {
```

Comment:
i don't think we need this function, let's remove it

Reply:
@frosty00, encoded params are required as without this function some graviex endpoints show signature is not a valid exception.